src/useConditionalEffect.ts

```diff
@@ -0,0 +1,12 @@
+import { useEffect, EffectCallback, DependencyList } from "react";
+
+const FALSE_DEP_ARRAY = [{}];
+const NOOP = () => {};
+
+export const useConditionalEffect = (
+  condition: boolean,
+  effect: EffectCallback,
+  dependencies: DependencyList | undefined
+) => {
+  useEffect(condition ? effect : NOOP, condition ? dependencies : FALSE_DEP_ARRAY);
```

Comment:
Pretty sure React will complain when changing the length of the dependencies, so might be worth doing something like this instead? Could also do it by globally caching the fake deps per array length, though this way does handle not having any, while caching would need specific handling

```suggestion
  const fakeDeps = useRef(() => dependencies && new Array(dependencies.length));

  useEffect(condition ? effect : NOOP, condition ? dependencies : fakeDeps.current);
```

Reply:
Good catch, fixed.